### PR TITLE
Clean up Navigation and Routes

### PR DESCRIPTION
**Pull request by caziz**
This cleans up lots of the navigation using the nav bar and buttons on pages. Pages now cleanly match appropriate routes and everything is linked together nicely. The nav bar is now always shown, but it is different depending on if someone is logged in.

### CODE REVIEW

**Review by DevenVimalPatel — APPROVED** [review]

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class UsersController < ApplicationController
-  before_action :correct_user, only: [:show, :edit, :update, :destroy]
+  before_action :correct_user, only: %i[edit update destroy]
+  before_action :set_user, only: %i[show edit update destroy]
```
Comment (awolk) [inline app/controllers/users_controller.rb:2]: This is backwards right? Also I think show should be protected

**Review by awolk — CHANGES_REQUESTED** [review]
Nice, just one small comment

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (caziz) [reply app/controllers/users_controller.rb:2]: I wanted to allow users to view other users profiles eg. you can now click on peoples profiles from the view resumes page

Comment (awolk) [reply app/controllers/users_controller.rb:2]: sgtm

**Pull request merged**